app/controllers/articles_controller.rb

```diff
@@ -0,0 +1,199 @@
+# frozen_string_literal: true
+
+#  News Articles
+#
+#  Actions
+#
+#    create:           Create article from data in "new" form
+#    destroy::         Destroy article
+#    edit::            Display form for editing article
+#    index::           List all articles
+#    index_articles::  List selected (based on last search) articles
+#    new::             Display form for new article
+#    show::            Show one article
+#    update::          Update article from "edit" form
+#
+#
+#  Callbacks and Methods
+#
+#    ignore_request_unless_permitted:: Unless user permitted to perform request,
+#                       then index_articles
+#    permitted?         boolean: permitted to create/update/destroy Articles
+#
+class ArticlesController < ApplicationController
+  before_action :login_required, except: [
+    :index,
+    :index_articles,
+    :list_articles, # aliased
+    :show,
+    :show_article # aliased
+  ]
+  before_action :store_location
+  before_action :ignore_request_unless_permitted, except: [
+    :index,
+    :index_articles,
+    :list_articles, # aliased
+    :show,
+    :show_article # aliased
+  ]
+
+  ############ - Actions to Display data (index, show, etc.)
+
+  # List selected articles, based on current Query.
+  def index_articles
+    query = find_or_create_query(:Article, by: params[:by])
+    show_selected_articles(
+      query,
+      id: params[:id].to_s,
+      always_index: true
+    )
+  end
+
+  # List all articles
+  def index
+    query = create_query(
+      :Article,
+      :all,
+      by: :created_at
+    )
+    show_selected_articles(query)
+  end
+
+  alias_method :list_articles, :index
+
+  # Display one Article
+  def show
+    return false unless (@article = find_or_goto_index(Article, params[:id]))
+
+    @canonical_url = "#{MO.http_domain}/articles/#{@article.id}"
+  end
+
+  alias_method :show_article, :show
+
+  ############ Actions to Display forms -- (new, edit, etc.)
+
+  def new
+    @article = Article.new
+  end
+
+  alias_method :create_article, :new
+
+  # Edit existing article
+  def edit
+    pass_query_params
+    @article = find_or_goto_index(Article, params[:id])
+  end
+
+  alias_method :edit_article, :edit
+
+  ############ Actions to Modify data: (create, update, destroy, etc.)
+
+  def create
+    if flash_missing_title?
+      render(:new)
+      return
+    else
+      @article = Article.new(
+        title: params[:article][:title],
+        body: params[:article][:body],
+        user_id: @user.id
+      )
+      @article.save
+      redirect_to article_path(@article.id)
+    end
+  end
+
+  def update
```

Comment:
@JoeCohen -- What do you do when codeclimate gets picky about ABC? I'm baffled by these, even though i guess i _vaguely_ get what it means. But it would be great if i understood...

Reply:
This appears to be resolved.